src/pause.sol

```diff
@@ -19,17 +19,21 @@ import {DSNote} from "ds-note/note.sol";
 import {DSAuth, DSAuthority} from "ds-auth/auth.sol";
 
 contract DSPause is DSAuth, DSNote {
-    // --- auth ---
-    function setOwner(address owner_) public {
-        require(msg.sender == address(proxy), "ds-pause-undelayed-ownership-change");
+    // --- admin ---
+    modifier wait() {
+        require(msg.sender == address(proxy), "ds-pause-undelayed-call"); _;
+    }
+    function setOwner(address owner_) public wait {
         owner = owner_;
         emit LogSetOwner(owner);
     }
-    function setAuthority(DSAuthority authority_) public {
-        require(msg.sender == address(proxy), "ds-pause-undelayed-authority-change");
+    function setAuthority(DSAuthority authority_) public wait {
```

Comment:
Do you know if there are practical benefits of using e.g. `address` here instead of a contract type?

Reply:
For inheritance and test usage cases I see as better practice to keep all the interfaces with real types. Then you can cast inside the function. But I'm not sure if in this case you can overwrite the params types if `DSAuth` is already receiving a `DSAuthority` in its functions.
I remember it wasn't possible to overwrite the returns value types.